cmd/cloud-network-config-controller/main.go

```diff
@@ -63,23 +68,33 @@ func main() {
 		klog.Exitf("Error building kubernetes clientset: %s", err.Error())
 	}
 
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: v1core.New(kubeClient.CoreV1().RESTClient()).Events("")})
+	eventRecorder := eventBroadcaster.NewRecorder(clientgoscheme.Scheme, corev1.EventSource{Component: component})
+
+	rl, err := resourcelock.New(
+		resourcelock.ConfigMapsLeasesResourceLock,
+		controllerNamespace,
+		resourceLockName,
+		kubeClient.CoreV1(),
+		kubeClient.CoordinationV1(),
+		resourcelock.ResourceLockConfig{
+			Identity:      controllerName,
+			EventRecorder: eventRecorder,
```

Comment:
EventRecorder is optional according to the API spec:
https://pkg.go.dev/k8s.io/client-go/tools/leaderelection/resourcelock#ResourceLockConfig

Any reason why you added this?

Reply:
No good reason. I removed it now. ;)